DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -271,6 +299,10 @@ protected override void ProcessFile(InstallFile file, XPathNavigator nav)
                 {
                     this._upgradeScript = file;
                 }
+                else if (file.Name.StartsWith("schema.", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    this._schemaScripts.Add(file);
+                }
                 else if (type.Equals("install", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
While we are in here, can we cleanup some of the logic to remove duplication and to make this more clear.

For example for the first conditional, make it check for either name starting with "install" or a type of install.

Also, while we are here is it appropriate that "the no match" is uninstall?  That seems a bit dangerous of a default.  I realize outside the scope of this a bit, but if we are editing it would be good to do this this way.

Reply:
Actually the type install is a bit of a bucket since it includes everything except uninstall. In hindsight the entire solution could have depended on correct type labeling. But at some point it all got mixed up.